packages/loot-core/src/server/accounts/export-to-csv.ts

```diff
@@ -75,23 +77,46 @@ export async function exportQueryToCSV(query) {
       .options({ splits: 'all' }),
   );
 
-  const parentsPayees = new Map();
+  // initialize a map to allow splits to have correct number of split from
+  const parentsChildCount = new Map();
   for (const trans of transactions) {
     if (trans.IsParent) {
-      parentsPayees.set(trans.Id, trans.Payee);
+      parentsChildCount.set(trans.Id, 0);
     }
   }
 
-  // filter out any parent transactions
-  const noParents = transactions.filter(t => !t.IsParent);
+  // sort order for children is negative number in their sort order, so invert when mapping
+  // count how many children each parent has by incrementing for each child found
+  const childSplitOrder = new Map();
+  for (const trans of transactions) {
+    if (trans.IsChild) {
+      childSplitOrder.set(trans.Id, trans.SortOrder * -1);
+      parentsChildCount.set(
+        trans.ParentId,
+        parentsChildCount.get(trans.ParentId) + 1,
+      );
+    }
+  }
 
-  // map final properties for export and grab the payee for splits from their parent transaction
-  const transactionsForExport = noParents.map(trans => {
+  // map final properties for export and grab the child count for splits from their parent transaction
+  const transactionsForExport = transactions.map(trans => {
```

Comment:
We need to be able to import this csv back into actual, and other systems, and have the total amount be correct.

Having the parents in here makes the data too incompatible in my opinion. For example if you open the csv in openoffice or office, the default sum for the amount column now no longer matches your account balance.

If we want the parent information, parent and child transactions need to use different columns for the amount. (or child transactions can have an extra columns with parent information).

Reply:
as there doesn't seem to be any shared standard to follow, i was thinking this way would allow it to be imported in most places, but would obviously require data-maintenance on the import, either removing the parent or somehow combining them.

maybe child transactions could have an "amount" and a "split_from_amount" ? we could then look at trying to merge transactions on csv-import that have same date and "split_from_amount"?

can implement once decided.

Reply:
@wdpk were you thinking of keeping the parent transaction in there? If so, I think it's the parent transaction that would be best with an extra coluimn. It's amount column should be 0, (so it doesn't contribute to the balance of any account amount, or category budget). But it can have another column that contains it's total. Maybe `split_amount` or something?

I think that would keep all the data in the csv, (for apps that wanted to process the data back into splits/parents), but ensures the balances are correct immediately after import.

Reply:
I think keeping the parent would be nice to have the "whole picture" as there doesn't seem to be an agreed-upon standard, so might as well shoot for the moon?

`split_amount` sounds like a good solution. Also yes, much better to have the import be generally right even if happens to include some "0" transactions if the import doesn't look at the splits. I can implement.

Thanks for the review, I will update this PR.

Reply:
Export now exports 0 for `amount` on parents, and adds a `Split_Amount` for parents that has their `amount`.